inst/tests/tests.Rraw

```diff
@@ -18425,3 +18425,9 @@ dt = data.table(a = 1L)
 test(2252.1, dt[, b:=2L], error = "\\[ was called on a data.table.*not data.table-aware.*':='")
 test(2252.2, dt[, let(b=2L)], error = "\\[ was called on a data.table.*not data.table-aware.*'let'")
 rm(.datatable.aware)
+
+# passing a function in env= doesn't trip up processing 'j', #6026
+DT=data.table(a=1:2, b=3:4)
+test(2253.1, DT[, builtin(b), env=list(builtin=sum)], 7L)
+test(2253.2, DT[, closure(b), env=list(closure=var)], 0.5)
```

Comment:
This test passes when I run it here but fails (Observed: could not find function "b") when I run `test.data.table()`, not sure why yet

Reply:
Ah, a red herring. `var="b"` was making me think `b` was the issue, but it was `var` being overwritten.